vsphere/resource_vsphere_guest_os_customization_test.go

```diff
@@ -0,0 +1,176 @@
+)
+
+func TestAccResourceVSpherGOSC_windows_basic(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("win")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCWindows(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSpherGOSC_windows_workGroup(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("win")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCWindowsAllPropsWorkGroup(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSpherGOSC_linux(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("lin")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCLinux(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSpherGOSC_sysprep(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("lin")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCWindowsPrep(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func testAccGOSCExists(resourceName string, goscName string, expectToExist bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resource := fmt.Sprintf("vsphere_guest_os_customization.%s", resourceName)
+		vars, err := testClientVariablesForResource(s, resource)
+		if err != nil {
+			return err
+		}
+		_, err = guestoscustomizations.FromName(vars.client, goscName)
+		if err != nil && expectToExist {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccGOSCWindows(resourceName string, goscName string) string {
+	return fmt.Sprintf(`
+		resource "vsphere_guest_os_customization" %q {
+			name = %q
+			type = "Windows"
+			spec {
+				windows_options {
+					computer_name = "Windows-Computer-Name"
```

Comment:
Should fail if > 15 characters.

```suggestion
					computer_name = "windows"
```